pkg/adaptation/result.go

```diff
@@ -751,6 +757,73 @@ func (r *result) adjustRlimits(rlimits []*POSIXRlimit, plugin string) error {
 	return nil
 }
 
+func (r *result) initAdjust() {
+	if r.reply.adjust == nil {
+		r.reply.adjust = &ContainerAdjustment{}
+	}
+}
+
+func (r *result) initAdjustAnnotations() {
+	r.initAdjust()
+	if r.reply.adjust.Annotations == nil {
+		r.reply.adjust.Annotations = map[string]string{}
+	}
+}
+
+func (r *result) initAdjustMounts() {
+	r.initAdjust()
+}
+
+func (r *result) initAdjustEnv() {
+	r.initAdjust()
+}
+
+func (r *result) initAdjustHooks() {
+	r.initAdjust()
+	if r.reply.adjust.Hooks == nil {
+		r.reply.adjust.Hooks = &Hooks{}
+	}
+}
+
+func (r *result) initAdjustLinux() {
+	r.initAdjust()
+	if r.reply.adjust.Linux == nil {
+		r.reply.adjust.Linux = &LinuxContainerAdjustment{}
+	}
+}
+
+func (r *result) initAdjustDevices() {
+	r.initAdjustLinux()
+}
+
+func (r *result) initAdjustResources() {
+	r.initAdjustLinux()
+	if r.reply.adjust.Linux.Resources == nil {
+		r.reply.adjust.Linux.Resources = &LinuxResources{
+			Memory:         &LinuxMemory{},
+			Cpu:            &LinuxCPU{},
+			HugepageLimits: []*HugepageLimit{},
+			Unified:        map[string]string{},
```

Comment:
We only need to init the fields that are pointers no ?

Reply:
Actually we need to init the map at least, maybe not the array

Reply:
Yes, maps need to be initialized if you might try adding or updating map entries.